### DISCUSSION TITLE
Prepare for a 4.1 release?

### DISCUSSION THREAD

**Discussion by gerhardol** [gitextensions/gitextensions#10833] [General]
There are some open PRs with issues in [vNext](https://github.com/gitextensions/gitextensions/milestone/65) that probably should be included, but should a 4.1 be considered?

There are no groundbreaking changes but a lot of tuning for 4.0 changes.
The improved handling in https://github.com/gitextensions/gitextensions/pull/10802 is an argument for a release.

**Comment by RussKie** [comment]
Yes, I think that's a good idea, and we should push out 4.1 sooner rather
than later.

**Reply by gerhardol** [reply]
I believe that there are PRs for all issues in the milestone that are seen as regressions that can be handled (no theming...)
Many of the open PRs should be included.

**Comment by RussKie** [comment]
Let's wrap everything we want to include, and I'll start preparing a
release.

**Comment by gerhardol** [comment]
For issues added to vNext, #10423 is in progress (#10852 #10854).
Other issues will likely not make it.

Some other PRs could be included (#10846 ?).
Time to make a rc?

**Reply by RussKie** [reply]
I created release/4.1 branch and bumped the master.

**Reply by gerhardol** [reply]
A temp build for documentation is here: https://git-extensions-documentation.readthedocs.io/en/feature-prepare-4.1/
PR for adjustments to the app as well as for doc itself to follow.

**Reply by RussKie** [reply]
Please target the release/4.1 branch for anything that goes into the release. Once the release is done we'll merge 4.1 into the master.

**Reply by mstv** [reply]
> I created release/4.1 branch and bumped the master.

> I prefer to have master and release branch in sync.

Let's release 4.1 after merging the ready bugfix PRs.
Next time, we should create the branch right before releasing. We use to not merge other PRs during the finalization phase.
(The renaming in 234e3b4e8001aa1985483d46a208a9e896fd2a0c - merged into the release branch - finally made testing vNext PRs a merging pain.)

**Reply by RussKie** [reply]
> The renaming in https://github.com/gitextensions/gitextensions/commit/234e3b4e8001aa1985483d46a208a9e896fd2a0c - merged into the release branch - finally made testing vNext PRs a merging pain.

Could you please elaborate?

**Reply by RussKie** [reply]
There are three pull requests remaining: 
 https://github.com/gitextensions/gitextensions/milestone/65

**Reply by RussKie** [reply]
* I merged release/4.1 into the master.
* I also belatedly realised that new release/4.1 branch meant that we need to setup new projects in Transifex and in SignPath, which is a major burden. The v4.1 is an interaction of 4.x stream, so we should have branched into the release/4.0... So I'm planning to merge the release/4.1 into the release/4.0 branch and continue the release from the release/4.0 branch.

**Reply by RussKie** [reply]
> Next time, we should create the branch right before releasing. We use to not merge other PRs during the finalization phase.

This is an eternal struggle, isn't it? 😃. In the previous releases we tried to finalise the work on the master, and it would drag on and on. This time around we appear to be significantly more judicious. 
I do ack and agree with your point on avoiding merging changes in parallel streams during the release.

**Reply by mstv** [reply]
It is.

The nuisance with vNext PRs was that I could not rebase them on the latest merged PRs because they were merged into the release branch. So I needed to resolve merge conflicts again and again or to manually cherry-pick.

Another drawback is that GH does not automatically close fixed issues.

**Comment by RussKie** [comment]
The release is cut. 
@gitextensions/git-extensions-team could you please help with https://github.com/gitextensions/gitextensions/releases/tag/untagged-3d4376ce55b58c1ef89d.

Thank you.

**Reply by RussKie** [reply]
The beta is here: https://ci.appveyor.com/project/gitextensions/gitextensions/builds/46925248/artifacts

**Reply by RussKie** [reply]
I uninstalled the previous version and installed the latest. The app is failing to start - it's trying to locate git, and even if pointed to it - terminates... I can't repro it from a debug build though. 
Is anyone else having this issue?

**Reply by mstv** [reply]
- Git Extensions 4.1.0.16654
- Build 6084cf43185d40298ccdb63dadbefdfe7165e274
- Git 2.39.2.windows.1 (recommended: 2.40.1 or later)
- Microsoft Windows NT 10.0.19045.0
- .NET 6.0.16
- DPI 96dpi (no scaling)
- **Portable: True**
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 7.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Reply by mstv** [reply]
> Is anyone else having this issue?

Yes, but only using the installed version - also after restart of the VM.

**Reply by mstv** [reply]
The portable folder contains additional files:
TestableIO.System.IO.Abstractions.dll
TestableIO.System.IO.Abstractions.Wrappers.dll
(UserPlugins\GitExtensions.PluginManager\*)

**Reply by mstv** [reply]
Adding TestableIO.System.* resolves the issue for me.

**Comment by RussKie** [comment]
Thank you for the investigation! I totally overlooked that the updated
package brought new additional assemblies.
It should be a quick fix.
🙏

**Reply by gerhardol** [reply]
The changelog was incorrectly changed too.

* 4.0.1 and 4.0.2 was removed completely
* 4 PRs included in 4.0 but not in the changelog there are added in 4.1. Incorrect but better than never? However, the actual links were removed from the doc. (In this case, reference style links should be replaced with regular links.)
![image](https://user-images.githubusercontent.com/6248932/235462162-e647444e-cdac-4b65-b662-514b93159903.png)

Edit: I removed the first 3 from 4.1 as they were included in 4.0 with incorrect milestone.
#9925 should be included though in 4.1 release notes.
